### ISSUE TITLE
[PAT] Unproper use of NoReturn type hint

### ISSUE THREAD

**Issue by JennyKmu** [pyCGNS/pyCGNS#148]
Hello,

I noticed in the PAT.cgnsutil module an improper use of the "NoReturn" typehint, which causes linters to incorrectly assume the corresponding function never returns.

![image](https://github.com/user-attachments/assets/394fc1a4-0651-4c78-ad15-b82b68423fdc)

This was noticed with version v6.1.4, and after checking the code for the tagged version v6.2.2 and th main branch, it has not been corrected.

The NoReturn type hint intends to signify that the function either neither returns, or always returns through an exception. For functions always returning None, then the typehint should be None. (See [typing.NoReturn documentation](https://docs.python.org/3/library/typing.html#typing.NoReturn))

For a function always returning None, the return type should be None.

For instance:

```python
def removeChildByName(parent: TreeNode, name: str) -> NoReturn: ...
```

should be declared as:

```python
def removeChildByName(parent: TreeNode, name: str) -> None: ...
```

**Comment by MicK7** [comment]
Thanks for the report.
Fixed in main branch

**Comment by JennyCamugli** [comment]
Sorry for coming back to it, but ``removeChildByName`` was not the only occurence of this issue. After checking, all occurences of ``NoReturn`` in ``PAT.cgnsutil`` should be replaced by ``None``, and ``NoReturn`` can be removed from the imports.

I haven't noticed any other use of NoReturn in the project.

Best regards,
Jenny

**Comment by MicK7** [comment]
I agree that other occurences of `NoReturn` can be replaced by `None` but not all
For instance  `removeNodeFromPath` should keep a `NoReturn`  hint

**Comment by MicK7** [comment]
Feel free to propose a PR.

**Comment by JennyCamugli** [comment]
I don't have access to push a new branch to this repo. I'll look into forking the repo (haven't done so before).

I'm not sure why``removeNodeFromPath`` would never return, I don't see where it would be equivalent to doing something then calling ``exit()`` or raising an exception in all cases.

**Comment by MicK7** [comment]
> Maybe there's a confusion, as not having a return statement does not mean there is ``NoReturn``, it means there's an implicit ``return None`` ?

You are right . I didn't read the spec fully. There is no reason to keep NoReturn.
